Super advanced technology

UndeadNappist opened the pull request:
-Adds the super advanced technology trait
-Reorganizes Legion, Vault, and BoS job pathing
-Cleans up some of my old code
-Automatically adds the TRAIT_TECHNOPHREAK to BoS and Vaulties
-Makes it so you can no-longer deconstruct autolathes at all
-Makes it so you must have the TRAIT_TECHNOPHREAK to unanchor the autolathe
-Adds the construction lathe, which is like the autolathe, but can neither produce Security items nor Contraband (Hacked) items
-Replaces the autolathe with the constructionlathe in R&D (sorry not sorry)
-constructionlathe you can fully disassemble
-Prerequisite TRAIT_TECHNOPHREAK to use power armor
-R&D machine now requires the technophreak trait to use
-Autolathe is straight-up indestructible.

UndeadNappist commented:
This has grown wildly out of my control and has turned into job refactoring.

LetterN commented on code/modules/jobs/job_types/security.dm:
```diff
@@ -187,93 +186,7 @@ Security Officer
 	outfit = /datum/outfit/job/security
 
 	access = list(ACCESS_SECURITY, ACCESS_SEC_DOORS, ACCESS_BRIG, ACCESS_COURT, ACCESS_MAINT_TUNNELS, ACCESS_MORGUE, ACCESS_WEAPONS, ACCESS_FORENSICS_LOCKERS, ACCESS_MINERAL_STOREROOM)
-	minimal_access = list(ACCESS_SECURITY, ACCESS_SEC_DOORS, ACCESS_BRIG, ACCESS_COURT, ACCESS_WEAPONS, ACCESS_MINERAL_STOREROOM) //BUT SEE /DATUM/JOB/WARDEN/GET_ACCESS()
```
`/* */` these codes instead of delet

UndeadNappist commented on code/game/objects/objs.dm:
```diff
@@ -242,3 +244,37 @@
 		icon_state = unique_reskin[choice]
 		to_chat(M, "[src] is now skinned as '[choice].'")
+
+/obj/proc/can_be_unfasten_wrench(mob/user, silent) //if we can unwrench this object; returns SUCCESSFUL_UNFASTEN and FAILED_UNFASTEN, which are both TRUE, or CANT_UNFASTEN, which isn't.
```
It's already obj/proc/unfasten. I'm just moving the proc to it's correct location in the file directory, see /code/game/machinery/_machinery.dm in the changes.

I can change it to only affect machinery, but it would likely mess up some structures or other things.

LetterN replied:
tbh it's just copypaste it's fine

AndrewL97 replied:
my reading comprehension is utter shite

UndeadNappist commented on code/modules/jobs/job_types/legion.dm:
```diff
@@ -1,9 +1,18 @@
 Needs whitelist
 */
-/datum/job/f13legate
+/datum/job/CaesarsLegion/Legionnaire/f13legate
```
Because /legion/ is already taken up by the lavaland job stuff for mining, inherited from /tg/ code. If you try to make it /legion/ it infinitely recursions.

We could remove the lavaland stuff.

UndeadNappist replied:
Also it's to differentiate between camp followers and legionnaires, who have caesar's but not /legionnaire/

UndeadNappist commented on code/modules/jobs/job_types/ncr.dm:
```diff
@@ -1,7 +1,7 @@
 Colonel
 */
-/datum/job/f13colonel
+/datum/job/NewCaliforniaRepublic/f13colonel
```
Same reason as legion above, /ncr/ is used for something somewhere. When you do that it becomes an infinite recursion.

Haven't traced as to where.

UndeadNappist replied:
Fixed.

UndeadNappist commented:
This is an enormous update with mapping and other things going on. I would very much appreciate someone else running tests on it too before it gets merged given how much it tinkers with.

-Autolathe is removed from NCR
-More monsters shoved directly outside of NCR and legion base to slow them down and stop rushing
-Constructionlathe is in and put into various places
-There are fewer autolathes (Den still has theirs')
-Legion armory lost a lot of ranged weaponry and ammo, gained bolas, shields, melee weapons, spears, handcuffs, rubbershot boxes of ammo, and more uniforms
-NCR armory got tweaked
-Gunstore is now more dangerous

May as well do all the mapping changes at once. Sling them to me if you've got them.

GhostEcho commented:
Gunstore could get a buff in loot if its gonna become more dangerous since its rather lacking from what I can see
The deathclaw cave to the east needs loot, and the deathclaw cavern down the ladder needs a buff in loot 
Also Id suggest making power armor a seperate perk than combing it with all this, knowing how to move in a power suit isnt really the same as knowing how to build machines

GhostEcho commented:
Also the cave to the northwest of caesers legion was a Yao Guai cave at one point, that could use some population

UndeadNappist commented:
Reopening, someone else can figure this out. Probably needs to be cleaned up.

UndeadNappist commented:
I've fixed all the conflicts and reverted and map changes.
I'll go after the map in a later balance update due to the way things have been changes (as it needs re-examining); for now, this just cleans up the job hierarchy.

In the future I will simply be using an automated text script for that job once I finish building it.

All other changes are implemented.

UndeadNappist commented:
So because this has gotten wildly out of control I'm going to provide a rundown of what all this shit DOES.
This does not touch the map. The map changes will come later after I have some time to carefully consider how the current map changes affect game balance.

1. Remove runtimes for calling on a null martial art
2. Makes autolathes indestructible and cannot be disassembled, only moved, and only by trait_technophreak (vault and BoS only)
3. Makes power armors only usable by those with the trait_technophreak
4. Reorganize and clean up job hierarchies to make them easier on people to edit them later (legion is now job/CaesarsLegion/f13campfollower and job/CaesarsLegion/Legionnaire/f13scout, for example)
5. Makes R&D console accessible to only those with trait_technophreak
6. Adds the constructionlathe to the game, which is just the autolathe but can be dismantled, moved, and does not have security or contraband items.
7. Makes autolathes non-constructable
8. Replaces the autolathe with the constructionlathe board in the R&D console
9. Makes it so if you try to spawn as a female legionnaire of any type, it sets you to male and randomizes your name.
10. Sets the number of BoS jobs correctly to what Ren stated he wanted them (which may or may not already be implemented)
11. Does NOT remove the autolathe circuit board from the game - it's just something you can't print anymore.
12. Moves some of the unfasten/deconn procs into the appropriate file and out of the incorrect one.
13. Fixed some edge cases where the autolathe would eat tools with the panel open.

GhostEcho commented on code/modules/jobs/job_types/bos.dm:
```diff
@@ -227,39 +203,30 @@ Scribe
+	jobtype = /datum/job/bos/f13initiateknight
 	suit = 			/obj/item/clothing/suit/armor/f13/combat/brotherhood
 	head = 			/obj/item/clothing/head/helmet/f13/combat/brotherhood
 	suit_store =	/obj/item/gun/ballistic/automatic/marksman/servicerifle
```
give them an aer9 instead of a service rifle, more thematically appropriate

UndeadNappist replied:
Done.

rhicora commented:
@UndeadNappist Let me know when this is ready to go!

UndeadNappist commented:
Entirely ready to be merged.
